### PR TITLE
nv2a: Match HW behavior when setting window clip

### LINKED ISSUE #907: Behavior when setting `NV097_SET_WINDOW_CLIP_*` does not match HW
### Bug Description

While debugging #735 I kept running into issues where surfaces had uninitialized data along edges. I wrote a [test case](https://github.com/abaire/nxdk_pgraph_tests/blob/main/src/tests/window_clip_tests.cpp) and found that xemu's behavior departs substantially from hardware for all the tests in the case that the test does not explicitly initialize all 8 clip regions. 

Compared to the [HW results](https://github.com/abaire/nxdk_pgraph_tests_golden_results/wiki/Results-Window_clip), on xemu, all of the exclusive mode clips fail to fill any of the region and all of the inclusive mode clips fill the entire region.

NOTE: When all clip regions are explicitly initialized (setting the unused ones to 0,0,0,0) the xemu output matches HW.

### Expected Behavior

On hardware, any set of a clip window updates the clip value for all following clip windows.

E.g., when setting clip0, then clip1, 2-7 will all have the value of clip1. Similarly, if setting clip1 then clip0, all clip values will be set to clip0. (Confirmed by reading back the NV_PGRAPH_WINDOWCLIPX0 and NV_PGRAPH_WINDOWCLIPY0 registers and by inverting ordering of clip rects).

[HW results](https://github.com/abaire/nxdk_pgraph_tests_golden_results/wiki/Results-Window_clip)

### xemu Version

Version:      0.6.6
Branch:       master
Commit:       c2e0067538fa5d24e57effce1adca172db6e8117
Date:         Mon May  9 02:38:25 AM UTC 2022

### System Information

CPU:          Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
OS Platform:  Linux
OS Version:   Ubuntu 21.10
Manufacturer: NVIDIA Corporation
GPU Model:    NVIDIA GeForce GTX 1070/PCIe/SSE2
Driver:       4.0.0 NVIDIA 470.103.01
Shader:       4.00 NVIDIA via Cg compiler

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by abaire**
Fixes #907 

[Test](https://github.com/abaire/nxdk_pgraph_tests/blob/main/src/tests/window_clip_tests.cpp)
[HW results](https://github.com/abaire/nxdk_pgraph_tests_golden_results/wiki/Results-Window_clip)

### CODE REVIEW

**Comment by abaire** [comment]
Found an additional bug in xemu wrt. dealing with clip regions larger than the surface size (there's a signed/unsigned mismatch, likely something else as well). Pulling that fix into this PR so the improved tests pass

Example of the bug: (the top line ends up at the bottom of the image)
Screenshot_20220509_071809

Depending on the specifics, this may also explain #800 as it's very similar behavior.

**Comment by abaire** [comment]
Looks like that wrapping bug is was the final display only (the surfaces emulating the guest are fine, the error is introduced in the last display blit) and has been fixed with the new UI.

File: hw/xbox/nv2a/pgraph.c [hunk 1]
```diff
@@ -1507,13 +1507,17 @@ DEF_METHOD(NV097, SET_WINDOW_CLIP_TYPE)
 DEF_METHOD_INC(NV097, SET_WINDOW_CLIP_HORIZONTAL)
 {
     int slot = (method - NV097_SET_WINDOW_CLIP_HORIZONTAL) / 4;
-    pg->regs[NV_PGRAPH_WINDOWCLIPX0 + slot * 4] = parameter;
+    for (; slot < 8; ++slot) {
+        pg->regs[NV_PGRAPH_WINDOWCLIPX0 + slot * 4] = parameter;
```
Comment (mborgerson) [inline hw/xbox/nv2a/pgraph.c:1511]: Nice discovery! I confirmed this with your test

**Review by mborgerson — CHANGES_REQUESTED** [review]
Please remove the updated submodule and I'll merge

**Comment by abaire** [comment]
> Please remove the updated submodule and I'll merge

Oops, fixed. My script to auto rebase my PR branches apparently doesn't account for submodule update properly.

**Pull request merged**